src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -1,15 +1,378 @@
+					{ "slug": "background", "name": "Background", "color": "" },
+				],
+			"supports": ["yith-wonder"]
+		};
+		setBackgroundColor();
+		setPrimaryColor();
+		setSecondaryColor();
+		setTertiaryColor();
+		setThemeColorPalette(colorStyle);
+
+		setSelectedColors(selectedColorsTemp);
+		currentData.data.palette[0] = selectedColorsTemp;
+		setCurrentOnboardingData(currentData);
+	};
+
+	const changeCustomPickerColor = async (color) => {
+		let primaryColorTemp = selectedColors?.color[0].color ?? '';
+		let secondaryColorTemp = selectedColors?.color[1].color ?? '';
+		let tertiaryColorTemp = selectedColors?.color[2].color ?? '';
+
+		let selectedColorsTemp = {
+			"slug": 'custom',
+			"name": 'Custom',
+			"color": [
+				{ "slug": "primary", "name": "Primary", "color": primaryColor ?? primaryColorTemp },
+				{ "slug": "secondary", "name": "Secondary", "color": secondaryColor ?? secondaryColorTemp },
+				{ "slug": "tertiary", "name": "Tertiary", "color": tertiaryColor ?? tertiaryColorTemp },
+				{ "slug": "background", "name": "Background", "color": backgroundColor ?? '' },
+			],
+			"supports": ["yith-wonder"]
+		};
+		
+		switch (colorPickerCalledBy) {
+			case 'background':
+					setBackgroundColor(color);
+					selectedColorsTemp.color[3].color = color;
+					break;
+			case 'primary':
+					setPrimaryColor(color);
+					selectedColorsTemp.color[0].color = color;
+					break;
+			case 'secondary':
+					setSecondaryColor(color);
+					selectedColorsTemp.color[1].color = color;
+ 					break;
+			case 'tertiary':
+					setTertiaryColor(color);
+					selectedColorsTemp.color[2].color = color;
+					break;
+		}
+		
+		setCustomColors();
+		setSelectedColors(selectedColorsTemp);
+		currentData.data.palette[0] = selectedColorsTemp;
+		setCurrentOnboardingData(currentData);		
+	}
+
+	const selectCustomColor = (colorType) => {
+		const showColorPickerTemp = showColorPicker;
+		setShowColorPicker(!showColorPickerTemp);
+
+		if (!showColorPickerTemp)
+			setColorPickerCalledBy(colorType);
+		else
+			setColorPickerCalledBy('');
+	}
+
+	function buildPalettes () {
+		let paletteRenderedList = [];
+		for (const colorStyle in colorPalettes) {
+			paletteRenderedList.push(
+				<div className={`color-palette ${colorStyle == selectedColors?.slug ? 'color-palette-selected' : ''} `}
+					onClick={(e) => handleClick(colorStyle)}>
+					<div className='color-palette__colors'>
+						<div className='color-palette__colors-tert'
+							style={{ backgroundColor: `${colorPalettes[colorStyle][0]}` }}/>
+						<div className='color-palette__colors-scnd'
+							style={{ backgroundColor: `${colorPalettes[colorStyle][1]}` }}/>
+						<div className='color-palette__colors-prim'
+							style={{ backgroundColor: `${colorPalettes[colorStyle][2]}` }} />
+					</div>
+					<div className='color-palette__name'>
+						{colorStyle?.charAt(0).toUpperCase() + colorStyle?.slice(1) }
+					</div>
+				</div>
+			);
+		}
+
+		return paletteRenderedList;
+	}
```

Comment:
These can go into a ColorPaletteList component that accepts a list of color palettes.